woocommerce/pinterest-for-woocommerce
# Product attribute for Google product category

### Changes proposed in this Pull Request:

This is a continuation of #303 which adds the basic framework for product attributes. This PR specifically adds a product attribute for setting the Google product category.

The Google product categories are downloaded  from the following location: https://www.google.com/basepages/producttype/taxonomy-with-ids.en-US.txt

In the Facebook extension this list was converted to a PHP class as there were issues with storing it in one transient and automatically downloading a new copy. So here we copied it in the same format including a generation script which can be found in the `bin` folder if we need to generate / bundle a new copy of the list.

Searching for a category reuses the AJAX functionality that is built into the WooCommerce product and category `select2` input fields. So typing in partial text will match any categories / subcategories. The results contain the subcategory and all it's parent categories as a complete string, ex:

> Apparel & Accessories > Clothing Accessories > Hats

This full string is put directly into the XML feed.

For variable products the Google categories can be set only on the parent product. So for variations we merge in the parent attributes so it can output the category for each variation product.

Resolves part of #195

### Detailed test instructions:

1. Update to this PR
2. Set the Google category in the Pinterest tab when editing several products (both simple and variable)
3. Regenerate the feed and view the XML to confirm it contains the Google category (Note: I haven't found a good way to speed up the regeneration of a feed besides waiting 24h)
```
<item>
    <g:id>58</g:id>
    <title>
        <![CDATA[ Small Plastic Hat ]]>
    </title>
    ....
    <g:google_product_category>Apparel & Accessories > Clothing Accessories > Hats</g:google_product_category>
</item>
```

### Changelog entry

* Add - Product attribute for Google product category.
